ingress-node-firewall/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
+type IngressNodeFirewallICMPRule struct {
+	// ICMPType define ICMP Type Numbers (RFC 792).
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum:=255
+	// +kubebuilder:validation:Minimum:=0
+	ICMPType uint8 `json:"icmpType,omitempty"`
+
+	// ICMPCode define ICMP Code ID (RFC 792).
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum:=255
+	// +kubebuilder:validation:Minimum:=0
+	ICMPCode uint8 `json:"icmpCode,omitempty"`
+}
+
+// IngressNodeFirewallProtoRule define ingress node firewall rule for TCP, UDP and SCTP protocols
+type IngressNodeFirewallProtoRule struct {
+	// Ports can be a single port or range of ports using start-end range format
+	// if admin wanted to filter single port then setting a single port as an integer for example Ports: 80
+	// if admin wanted to filter range of ports then using "start-end" range string format for example Ports: "80-100"
```

Comment:
Just want to tidy this up a little, for example, you should use the json tag name not the go field name, as users interact with the json names only
```suggestion
	// ports defines either a single port or a range of ports to apply a protocol rule too.
	// To filter a single port, set a single port as an integer value. For example ports: 80.
	// To filter a range of ports, use a "start-end" range, string format. For example ports: "80-100".
```